### PR TITLE
Updated to Swift 3 and fixes

### PR DESCRIPTION
**Pull request by JEKos**
Updated to Swift 3 and application.registerForRemoteNotifications() add to iOS 10 too

### CODE REVIEW

**Comment by JEKos** [comment]
I signed it!

**Comment by mcdonamp** [comment]
Thanks @JEKos, especially given how painful some migrations to Swift 3 have been! I don't think we've settled whether we're going to use Swift 3/XCode 8 by default for all new code, so we'll wait until we've synced internally to merge this. It might be fun to fork this into a Swift 3.0 scheme and keep it in parallel to the Swift 2.x one for devs on both (though TBH we're really not sure how much 2.x code exists in the world vs 3.x--assume it's a fair margin of existing development).

**Comment by JEKos** [comment]
ok, but the `application.registerForRemoteNotifications()` for iOS 10 is important, notification doesn't work in iOS 10 without this line

**Comment by morganchen12** [comment]
cc @iulukaya

**Comment by roughike** [comment]
@mcdonamp What @JEKos said. The Swift sample is broken for iOS 10, as notifications don't come through when the app is in background. This was a major headache for our team.

[Here's a PR that fixes that, and that only](https://github.com/firebase/quickstart-ios/pull/101)

See also:
* https://github.com/firebase/quickstart-ios/issues/102
* https://github.com/firebase/quickstart-ios/issues/103

**Comment by mcdonamp** [comment]
Cool, merged #101 (which now caused conflicts here), so hopefully people are unblocked on iOS 10 while we chat about Swift 3. Given that Swift 3 means XCode 8, and thus means most people will be using iOS 10, it's probably reasonable that we make this change, I think we're just trying to make sure we can update some of our other material (like our docs) to Swift 3 so it's consistent.

File: messaging/FCMSwift/AppDelegate.swift [hunk 1]
```diff
@@ -16,124 +16,127 @@
 
 import UIKit
```
Comment (qkies) [inline messaging/FCMSwift/AppDelegate.swift]: UIKit

File: messaging/FCMSwift/AppDelegate.swift [hunk 2]
```diff
@@ -16,124 +16,127 @@
...
 import UserNotifications
-
 import Firebase
 import FirebaseInstanceID
```
Comment (qkies) [inline messaging/FCMSwift/AppDelegate.swift]: UIKit

File: messaging/FCMSwift/AppDelegate.swift [hunk 3]
```diff
@@ -16,124 +16,127 @@
 
 import UIKit
 import UserNotifications
-
 import Firebase
```
Comment (qkies) [inline messaging/FCMSwift/AppDelegate.swift]: UIKit

**Comment by iulukaya** [comment]
I think we addressed the remaining part in our swift 3 upgrade. Thanks for your contribution.

**Pull request closed without merging**